### ISSUE TITLE
Login Problem after upgrading

### ISSUE THREAD

**Issue by fischhase** [owncloud/core#27742]
Hallo dear owncloud team,

thanks for your great work. I am upgrading my owncloud-installation from 7 to 9. It worked fine from 7 to 8.1, but from 8.1 to 8.2 and from 8.1 directly to 9.1.5.2. I could not login anymore. 

### Steps to reproduce
1. Uprade via button in the administrator 
or installing a working 9.1.5.2-Installation and overwriting the owncloud.db. After that, the installtion upgrades the DB by itself
2. Try to log in as administrator
3. getting the message: "Falsches Passwort. Soll es zurückgesetzt werden?" (Wrong password. Should it been replaced?)
4. Same with user passwords
5. Also Caldav-Login does not work

### Expected behaviour
I would expect a working login  :-)

### Actual behaviour
see above: Login Loop

### Server configuration
**Operating system**:
Debian (I think). It is a Hetzner Managed Server

**Web server:**
Apache 
**Database:**
SQLite3

**PHP version:**
7 
**ownCloud version:** (see ownCloud admin page)
9.1.5.2
**Updated from an older ownCloud or fresh install:**
update

**Where did you install ownCloud from:**
https://download.owncloud.org/community/owncloud-9.1.5.tar.bz2

**Signing status (ownCloud 9.0 and above):**
?
```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results into https://gist.github.com/ and puth the link here.
```
only the login screen appears

**The content of config/config.php:**
<?php
$CONFIG = array (
  'instanceid' => 'deleted',
  'passwordsalt' => 'deleted',
  'secret' => 'deleted',
  'trusted_domains' =>
  array (
    0 => 'cloud.meinedomain.de',
  ),
  'datadirectory' => '/deleted/owncloud/data',
  'overwrite.cli.url' => 'https://cloud.meinedomain.de',
  'dbtype' => 'sqlite3',
  'version' => '9.1.5.2',
  'logtimezone' => 'UTC',
  'installed' => true,
  'theme' => '',
  'loglevel' => 1,
  'maintenance' => false,
  'trashbin_retention_obligation' => 'auto',
);

```
Log in to the web-UI with an administrator account and click on
'admin' -> 'Generate Config Report' -> 'Download ownCloud config report'
This report includes the config.php settings, the list of activated apps
and other details in a well sanitized form.

or 

If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder
I would do that, but sudo is not allowed on meanaged server. Any Tipps to run occ-scripts on managed servers?

*ATTENTION:* Do not post your config.php file in public as is. Please use one of the above
methods whenever possible. Both, the generated reports from the web-ui and from occ config:list
consistently remove sensitive data. You still may want to review the report before sending.
If done manually then it is critical for your own privacy to dilligently
remove *all* host names, passwords, usernames, salts and other credentials before posting.
You should assume that attackers find such information and will use them against your systems.
```

**List of activated apps:**
contact
calendar
plus standard apps
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder.
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no (just SSL)
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser: opera an firefox**

**Operating system:**
Windows 10
#### Web server error log
```
Insert your webserver log here
I do have access to logs a day later :-(

#### ownCloud log (data/owncloud.log)
```{"reqId":"Xm9ZC7WXyNypecF7vqDv","remoteAddr":"31.16.227.147","app":"core","message":"Login failed: 'fischhase' (Remote IP: '31.16.227.147')","level":2,"time":"2017-04-25T12:22:53+00:00","method":"POST","url":"\/index.php\/login?user=fischhase","user":"--"}
{"reqId":"j8y\/S8QVS7GSASXEEteO","remoteAddr":"31.16.227.147","app":"core","message":"Login failed: 'martin' (Remote IP: '31.16.227.147')","level":2,"time":"2017-04-25T12:23:44+00:00","method":"POST","url":"\/index.php\/login?user=fischhase","user":"--"}

```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
QMIGRATE: Migrate is installed, version 1.4.0
jquery.js:8630XHR finished loading: GET "https://cloud.fischhase.de/core/img/breadcrumb.svg".
send @ jquery.js:8630
ajax @ jquery.js:8166
SVGSupport.checkMimeType @ js.js?v=b1e565e…:1368
initCore @ js.js?v=b1e565e…:1527
j @ jquery.js:3099
fireWith @ jquery.js:3211
ready @ jquery.js:3417
I @ jquery.js:3433
jquery.js:8630 XHR finished loading: GET "https://cloud.fischhase.de/cron.php".
send @ jquery.js:8630
ajax @ jquery.js:8166
n.(anonymous function) @ jquery.js:8311
(anonymous) @ backgroundjobs.js?v=b1e565e…:24
j @ jquery.js:3099
fireWith @ jquery.js:3211
ready @ jquery.js:3417
I @ jquery.js:3433

b) The network log 
don't know how to do this
c) ...
```
Thanks 
Maritn

**Comment by kdslkdsaldsal** [comment]
> thanks for your great work. I am upgrading my owncloud-installation from 7 to 9. It worked fine from 7 to 8.1, but from 8.1 to 8.2 and from 8.1 directly to 9.1.5.2. I could not login anymore.

As this currently doesn't contain the information how the upgrade was done in detail make sure that you have done the following upgrade path:

your 7.0.x -> latest 7.0.x -> latest 8.0.x -> latest 8.1.x -> latest 8.2.x -> latest 9.0.x -> latest 9.1.x

Before each upgrade step make sure that you have deleted all files and folders except /data/ and /config/. After running the upgrade routine also make sure that you have cleared the browser cache as well as any webserver cache (restarting webserver and possible php-fpm processes).

Everything else won't work and might show issues.

**Comment by fischhase** [comment]
Hallo,

thanks for your answer. Indeed, I tried ist several times with different 
ways. At the end I installed every version as new and activated contact 
and calendar. Then I copied data, owncloud and config.php from the 
former version. So I always had the chance to redo a step. At the end, 
it worked with nearly no problems. Only two calenders were lost 
somewhere between 8.1 and 9. Anyway. I could export and import them.

Thanks and bye
Martin


Am 25.04.2017 um 15:21 schrieb kdslkdsaldsal:
